Cannot install v3.0.0-post from GitHub URL

* Puppeteer version: v3.0.0-post
* Platform / OS version: Windows 10 x64
* URLs (if applicable):
* Node.js version: v14.0.0-v8-canary20200407ca0d7f295b

```
j:\temp>npm install -g https://github.com/GoogleChrome/puppeteer
(node:3008) [DEP0139] DeprecationWarning: Calling process.umask() with no arguments is prone to race conditions and is a potential security vulnerability.
(node:1764) [DEP0139] DeprecationWarning: Calling process.umask() with no arguments is prone to race conditions and is a potential security vulnerability.

> puppeteer@3.0.0-post install C:\Users\vmb\AppData\Roaming\npm\node_modules\puppeteer
> node install.js

Compiling TypeScript before install...
Error running TypeScript Error: Command failed: npm run tsc
(node:1288) [DEP0139] DeprecationWarning: Calling process.umask() with no arguments is prone to race conditions and is a potential security vulnerability.
(node:1144) [DEP0139] DeprecationWarning: Calling process.umask() with no arguments is prone to race conditions and is a potential security vulnerability.
'tsc' is not recognized as an internal or external command,
operable program or batch file.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! puppeteer@3.0.0-post tsc: `tsc --version && tsc -p . && cp src/protocol.d.ts lib/ && cp src/externs.d.ts lib/`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the puppeteer@3.0.0-post tsc script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\vmb\AppData\Roaming\npm-cache\_logs\2020-04-16T14_48_13_944Z-debug.log

    at ChildProcess.exithandler (child_process.js:303:12)
    at ChildProcess.emit (events.js:315:20)
    at maybeClose (internal/child_process.js:1051:16)
    at Socket.<anonymous> (internal/child_process.js:442:11)
    at Socket.emit (events.js:315:20)
    at Pipe.<anonymous> (net.js:651:12) {
  killed: false,
  code: 1,
  signal: null,
  cmd: 'npm run tsc',
  stdout: '\n' +
    '> puppeteer@3.0.0-post tsc C:\\Users\\vmb\\AppData\\Roaming\\npm\\node_modules\\puppeteer\n' +
    '> tsc --version && tsc -p . && cp src/protocol.d.ts lib/ && cp src/externs.d.ts lib/\n' +
    '\n',
  stderr: '(node:1288) [DEP0139] DeprecationWarning: Calling process.umask() with no arguments is prone to race conditions and is a potential security vulnerability.\n' +
    '(node:1144) [DEP0139] DeprecationWarning: Calling process.umask() with no arguments is prone to race conditions and is a potential security vulnerability.\n' +
    "'tsc' is not recognized as an internal or external command,\r\n" +
    'operable program or batch file.\r\n' +
    'npm ERR! code ELIFECYCLE\n' +
    'npm ERR! errno 1\n' +
    'npm ERR! puppeteer@3.0.0-post tsc: `tsc --version && tsc -p . && cp src/protocol.d.ts lib/ && cp src/externs.d.ts lib/`\n' +
    'npm ERR! Exit status 1\n' +
    'npm ERR! \n' +
    'npm ERR! Failed at the puppeteer@3.0.0-post tsc script.\n' +
    'npm ERR! This is probably not a problem with npm. There is likely additional logging output above.\n' +
    '\n' +
    'npm ERR! A complete log of this run can be found in:\n' +
    'npm ERR!     C:\\Users\\vmb\\AppData\\Roaming\\npm-cache\\_logs\\2020-04-16T14_48_13_944Z-debug.log\n'
}
npm WARN You are using a pre-release version of node and things may not work as expected

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! puppeteer@3.0.0-post install: `node install.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the puppeteer@3.0.0-post install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\vmb\AppData\Roaming\npm-cache\_logs\2020-04-16T14_48_14_217Z-debug.log
```

--- a/install.js
+++ b/install.js
@@ -14,45 +14,6 @@
  * limitations under the License.
  */
 
-const fs = require('fs');
-const path = require('path');
-const child_process = require('child_process');
-const {promisify} = require('util');
-
-const fsAccess = promisify(fs.access);
-const exec = promisify(child_process.exec);
-
-const fileExists = async filePath => fsAccess(filePath).then(() => true).catch(() => false);
-
-/*
- * Now Puppeteer is built with TypeScript, we need to ensure that
- * locally we have the generated output before trying to install.
- *
- * For users installing puppeteer this is fine, they will have the
- * generated lib/ directory as we ship it when we publish to npm.
- *
- * However, if you're cloning the repo to contribute, you won't have the
- * generated lib/ directory so this script checks if we need to run
- * TypeScript first to ensure the output exists and is in the right
- * place.
- */
-async function compileTypeScript() {
-  return exec('npm run tsc').catch(err => {
-    console.error('Error running TypeScript', err);
-    process.exit(1);
-  });
-}
-
-async function ensureLibDirectoryExists() {
-  const libPath = path.join(__dirname, 'lib');
-  const libExists = await fileExists(libPath);
-  if (libExists) return;
-
-  logPolitely('Compiling TypeScript before install...');
-  await compileTypeScript();
-}
-
-
 /**
  * This file is part of public API.
  *
@@ -62,13 +23,16 @@ async function ensureLibDirectoryExists() {
  * still possible to install a supported browser using this script when
  * necessary.
  */
+
+const compileTypeScriptIfRequired = require('./typescript-if-required');
+
 const supportedProducts = {
   'chrome': 'Chromium',
   'firefox': 'Firefox Nightly'
 };
 
 async function download() {
-  await ensureLibDirectoryExists();
+  await compileTypeScriptIfRequired();
 
   const downloadHost = process.env.PUPPETEER_DOWNLOAD_HOST || process.env.npm_config_puppeteer_download_host || process.env.npm_package_config_puppeteer_download_host;
   const puppeteer = require('./index');

--- a/package.json
+++ b/package.json
@@ -18,6 +18,7 @@
     "debug-unit": "node --inspect-brk test/test.js",
     "test-doclint": "mocha --config mocha-config/doclint-tests.js",
     "test": "npm run tsc && npm run lint --silent && npm run coverage && npm run test-doclint && npm run test-types",
+    "prepare": "node typescript-if-required.js",
     "prepublishOnly": "npm run tsc",
     "dev-install": "npm run tsc && node install.js",
     "install": "node install.js",

--- a/typescript-if-required.js
+++ b/typescript-if-required.js
@@ -56,6 +56,4 @@ async function compileTypeScriptIfRequired() {
 if (require.main === module)
   compileTypeScriptIfRequired();
 
-
 module.exports = compileTypeScriptIfRequired;
-

--- a/typescript-if-required.js
+++ b/typescript-if-required.js
@@ -1,0 +1,61 @@
+/**
+ * Copyright 2020 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const child_process = require('child_process');
+const path = require('path');
+const fs = require('fs');
+const {promisify} = require('util');
+
+const exec = promisify(child_process.exec);
+const fsAccess = promisify(fs.access);
+
+const fileExists = async filePath => fsAccess(filePath).then(() => true).catch(() => false);
+/*
+
+ * Now Puppeteer is built with TypeScript, we need to ensure that
+ * locally we have the generated output before trying to install.
+ *
+ * For users installing puppeteer this is fine, they will have the
+ * generated lib/ directory as we ship it when we publish to npm.
+ *
+ * However, if you're cloning the repo to contribute, you won't have the
+ * generated lib/ directory so this script checks if we need to run
+ * TypeScript first to ensure the output exists and is in the right
+ * place.
+ */
+async function compileTypeScript() {
+  return exec('npm run tsc').catch(err => {
+    console.error('Error running TypeScript', err);
+    process.exit(1);
+  });
+}
+
+async function compileTypeScriptIfRequired() {
+  const libPath = path.join(__dirname, 'lib');
+  const libExists = await fileExists(libPath);
+  if (libExists) return;
+
+  console.log('Puppeteer:', 'Compiling TypeScript...');
+  await compileTypeScript();
+}
+
+// It's being run as node typescript-if-required.js, not require('..')
+if (require.main === module)
+  compileTypeScriptIfRequired();
+
+
+module.exports = compileTypeScriptIfRequired;
+

